src/node/internal/internal_net.ts

```diff
@@ -1753,3 +1749,529 @@ export class SocketAddress implements _SocketAddress {
     return undefined;
   }
 }
+
+// ======================================================================================
+
+// Note: the bulk of the following code was authored by claude...
+// The implementation in Node.js is split between javascript and C++.
+// Here we do the entire implementation as TypeScript simply because
+// we don't need the C++ parts at all.
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+// IPv4/IPv6 CIDR network utilities
+function parseIPv4(ip: string): number[] {
```

Comment:
maybe add a comment about the behavior when the format is bad (returns an empty list)